doc/Programming-guide/Key-platform-features/Events/index.md

```diff
@@ -4,4 +4,854 @@ title: Events
+
+## Kaa Events SDK API
+
+#### Attach endpoint to user
+
+To get the details please visit [Owner verifiers]({{root_url}}Customization-guide/Customizable-system-components/Owner-verifiers/) page.
+
+#### Get ECF factory and create ECF object
+
+To access the Kaa event functionality, the client should implement the two following blocks of code.
+
+**Get ECF factory from Kaa:**
+
+<ul class="nav nav-tabs">
+  <li class="active"><a data-toggle="tab" href="#java1">Java</a></li>
+  <li><a data-toggle="tab" href="#cpp1">C++</a></li>
+  <li><a data-toggle="tab" href="#objc1">Objective-C</a></li>
+</ul>
+
+<div class="tab-content">
+<div id="java1" class="tab-pane fade in active" markdown="1" >
+
+```java
+import org.kaaproject.kaa.client.event.EventFamilyFactory;
+ 
+EventFamilyFactory eventFamilyFactory = kaaClient.getEventFamilyFactory();
+```
+
+</div><div id="cpp1" class="tab-pane fade" markdown="1" >
+
+```c++
+#include <kaa/event/gen/EventFamilyFactory.hpp>
+ 
+...
+EventFamilyFactory& eventFamilyFactory = kaaClient->getEventFamilyFactory();
+```
+
+</div><div id="objc1" class="tab-pane fade" markdown="1" >
+
+```objective-c
+#import <Kaa/Kaa.h>
+ 
+...
+ 
+EventFamilyFactory *eventFamilyFactory = [kaaClient getEventFamilyFactory];
+```
+
+</div>
+</div>
+
+**Get specific ECF object from ECF factory:**
+
+<ul class="nav nav-tabs">
+  <li class="active"><a data-toggle="tab" href="#java2">Java</a></li>
+  <li><a data-toggle="tab" href="#cpp2">C++</a></li>
+  <li><a data-toggle="tab" href="#objc2">Objective-C</a></li>
+</ul>
+
+<div class="tab-content">
+<div id="java2" class="tab-pane fade in active" markdown="1" >
+
+```java
+import org.kaaproject.kaa.demo.smarthouse.thermo.ThermoEventClassFamily;
+ 
+ThermoEventClassFamily tecf = eventFamilyFactory.getThermoEventClassFamily();
+```
+
+</div><div id="cpp2" class="tab-pane fade" markdown="1" >
+
+```c++
+#include <kaa/event/gen/ThermoEventClassFamily.hpp>
+ 
+...
+ThermoEventClassFamily& tecf = eventFamilyFactory.getThermoEventClassFamily();
+```
+
+</div><div id="objc2" class="tab-pane fade" markdown="1" >
+
+```objective-c
+#import<Kaa/Kaa.h>
+ 
+...
+ 
+ThermostatEventClassFamily *tecf = [self.eventFamilyFactory getThermostatEventClassFamily];
+```
+
+</div>
+</div>
+
+#### Send events
+
+To send one or more events, the client should proceed as described in this section.
+
+**Get endpoint addresses**
+
+Execute the asynchronous findEventListeners method to request a list of the endpoints supporting all specified EC FQNs (FQN stands for fully qualified name).
+
+<ul class="nav nav-tabs">
+  <li class="active"><a data-toggle="tab" href="#java3">Java</a></li>
+  <li><a data-toggle="tab" href="#cpp3">C++</a></li>
+  <li><a data-toggle="tab" href="#c3">C</a></li>
```

Comment:
Please, report the issue to the designer. I believe, the design should ensure minimal tab width. Correct me if I'm wrong.